featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -516,16 +516,27 @@ def _build_transform_features(self, all_features, entity, max_depth=0,
             input_types = groupby_prim.input_types[:]
             # if multiple input_types, only use first one for DFS
             if type(input_types[0]) == list:
-                input_types = input_types[0]
-            input_types.append(Id)
-
+                input_types = input_types[0][:]
+            id_is_input = Id in input_types
```

Comment:
I think we can simplify this whole for loop considerably now that we find available `Id` features directly.  
* We don't need to add an `Id` variable to the `input_types` to find possible `groupby` features anymore, since we already have a list of `Id` features: `id_matches`. 
* This removes the need for logic to handle whether the `input_types` has an `Id` feature or not.  
* Since we aren't adding an `Id` feature to `input_types` anymore, we don't need to split out the groupby variable using `matching_input[:-1]` / `matching_input[-1]`.